src/ILCompiler.ReadyToRun/src/Compiler/ReadyToRunMetadataFieldLayoutAlgorithm.cs

```diff
@@ -741,12 +743,12 @@ public static bool IsManagedSequentialType(TypeDesc type)
             {
                 foreach (FieldDesc field in type.GetFields())
                 {
-                    if (!field.IsStatic && !field.IsLiteral)
+                    if (!field.IsStatic &&
+                        !field.IsLiteral &&
+                        !field.FieldType.IsEnum &&
+                        !IsManagedSequentialType(field.FieldType))
```

Comment:
Nit: `FieldType` is a virtual property - I tend to extract it into a local when it's accessed multiple times.

Reply:
After the refactoring there is now just one call to FieldType in the code.